tests/test_transpiler_router.py

```diff
@@ -186,6 +186,63 @@ def test_routing_with_measurements():
     )
 
 
+def test_sabre_looping():
```

Comment:
Is this test checking that the looping problem of sabre is fixed by the shortest path approach? The final `assert > 50` might be ok in the majority of the cases, however, I see that a bit risky. If possible, I would try to fix a setting (by fixing the seed most likely) where we have complete control over all the moving parts: make sure the standard sabre get stuck in the loop, know in advance the precise number of swaps you get with a given threshold, etc... 
Or keep this random but, for testing, check that the number of swaps inserted with the old sabre (`threshold = inf`) is larger than the new one (`threshold=1.5`) rather than just comparing to an arbitrary value 50.
I would say that the optimal solution is actually having both the tests, the fixed one and the random one, if possible.

Reply:
Thanks. Please review the update.